packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-gitprovider.service.ts

```diff
@@ -142,6 +144,21 @@ export class AppLauncherGitproviderService implements GitProviderService {
     );
   }
 
+  getDetectedBuildRuntime(repoUrl: string): Observable<BuildTool> {
+    const url = this.detectorApiUrl + 'detect/build/' + repoUrl;
+    if (this.headers.has('X-app')) {
+      this.headers = this.headers.delete('X-app');
```

Comment:
Why are you deleting these headers?
Does this return a copy of the headers object or does it mutate the headers object? If it mutates, this will negatively affect the other service calls using these headers.